tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -208,13 +200,34 @@ export function getBinaryOpString(
       default:
         continue;
     }
+
+    let isNaN: string;
+    let dTypeN: string;
+    let boolN: string;
+    if (useVec4) {
+      isNaN = 'isnanVec4';
+      dTypeN = 'vec4<f32>';
+      boolN = 'vec4<bool>';
+    } else {
+      isNaN = 'isnan';
+      dTypeN = 'f32';
+      boolN = 'bool';
+    }
+
     return `
+      let aIsNaN = ${isNaN}(a);
+      let aPostLegalization = select(a, ${dTypeN}(42), aIsNaN);
```

Comment:
It seems we don't need aPostLegalization at all.

Reply:
My later patches will depend on `dTypeN(0)`, but `dTypeN` is not available in the global scope. One workaround is to use `a - a` instead, so it's critical to erase NaNs in `a` beforehand.

Reply:
I prefer you remove `aPostLegalization` and `bPostLegalization` in this PR. And use `a - a` in your needed PRs.

Reply:
Sorry I can't fully grab your idea. Can you show me an example of the unified `NOT_EQUAL`?